parcels/particle.py

```diff
@@ -251,6 +254,26 @@ def remove(self, indices):
         self.particles = np.delete(self.particles, indices)
         return particles
 
+    def density(self, lon=None, lat=None):
+        if lon is None:
+            lon = self.grid.U.lon
+        if lat is None:
+            lat = self.grid.U.lat
+        Density = np.zeros((lon.size, lat.size), dtype=np.float32)
+        Area = np.zeros(np.shape(Density), dtype=np.float32)
+
+        dx = (lon[1] - lon[0]) * 1852 * 60 * np.cos(lat*math.pi/180)
+        dy = (lat[1] - lat[0]) * 1852 * 60
+        for y in range(len(lat)):
+            Area[0:len(lon), y] = dy * dx[y]
+
+        # For each particle, find closest vertex in x and y
+        for p in self.particles:
+            Density[np.argmin(np.abs(p.lon - lon)), np.argmin(np.abs(p.lat - lat))] += 1
+        # Scale by cell area
```

Comment:
This is quite a computing-intensive operation. For one, don't we know `xi` and `yi` for each particle already? So I think we can replace `np.argmin(np.abs(p.lon - lon))` with `np.xi`, right?

And also, we don't need to recalculate `Area` on every call of `density()`; it is not going to change. Perhaps calculate only on first call of `density() `and then store it in the `grid` class?